### PR TITLE
Add validation to validate that floating points are finite numbers

### PR DESCRIPTION
**Pull request by Corniel**
As `double`'s (and `float`'s) can be assigned as `NaN`, `PositiveInfinity`, and `NegativeInfinity`, it useful to add validation that disallows these values if desired.

- [x] FluentValidation
- [x] DataAnnotations 
- [x] documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,86 +181,8 @@ for communicating that a service was unavailable. A use case for this can be to
...
-``` C#
-public class CustomValidator : AbstractValidator<Model>
-{
-    public CustomValidator()
-    {
-        RuleFor(m => m.PostalCode).ValidFor(m => m.Country);
-    }
-}
-```
+Provides a Fluent Valdation based implentation of the `Qowaiv.Validation.Abstractions.IValidator`
```
Comment (LauraXiulan) [inline README.md:184]: validation

Comment (LauraXiulan) [inline README.md:184]: implementation

File: specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+public class Is_invalid_for
+{
+    [TestCase(float.NaN)]
+    [TestCase(float.NegativeInfinity)]
+    [TestCase(float.PositiveInfinity)]
+    [TestCase(double.NaN)]
+    [TestCase(double.NegativeInfinity)]
+    [TestCase(double.PositiveInfinity)]
+    public void not_finate(object value)
```
Comment (LauraXiulan) [inline specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs:32]: finite

File: specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        => new IsFiniteAttribute().IsValid(value).Should().BeFalse();
+
+    [TestCase("Hello")]
+    [TestCase(42)]
+    [TestCase(128L)]
+    [TestCase(1024UL)]
+    public void non_floating_poitns(object value)
```
Comment (LauraXiulan) [inline specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs:39]: points

File: specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+       => new IsFiniteAttribute().IsValid(value).Should().BeFalse();
+}
+public class With_message
+{
+    [TestCase("nl", "De waarde van het Total veld is geen eindig getal.")]
+    [TestCase("en", "The value of the Total field is not a finate number.")]
```
Comment (LauraXiulan) [inline specs/Qowaiv.Validation.Specs/DataAnnotations/Attributes/Is_finite_specs.cs:45]: finite

File: src/Qowaiv.Validation.DataAnnotations/Attributes/MultipleOfAttribute.cs [hunk 5]
```diff
@@ -21,52 +21,39 @@ private MultipleOfAttribute(decimal factor)
...
-        Amount amount => IsValid((decimal)amount),
-        Money money => IsValid((decimal)money.Amount),
-        Percentage percentage => IsValid(100 * (decimal)percentage),
-        IConvertible convertible => IsValid(convertible),
-
-        _ => false,
-    };
+    public override bool IsValid(object? value)
+        => value is null
+        || value switch
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.DataAnnotations/Attributes/MultipleOfAttribute.cs:33]: Why did you remove string?

File: src/Qowaiv.Validation.DataAnnotations/README.md [hunk 6]
```diff
@@ -105,9 +105,21 @@ public class Model
 }
 ```
 
+### Is finite
+The `[IsFinite]` attribute validates that the floating point value of the field
+represents a finite (e.a. not NaN, of infinity).
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.DataAnnotations/README.md:110]: 'or' instead of 'of'

File: src/Qowaiv.Validation.Fluent/Extensions/System.FloatingPoints.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+#if NETSTANDARD
+        => !double.IsNaN(d) && !double.IsInfinity(d);
+#else
+        => double.IsFinite(d);
+#endif
+
+    [Pure]
+    public static bool IsFinite(this float f)
+#if NETSTANDARD
+    => !float.IsNaN(f) && !float.IsInfinity(f);
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.Fluent/Extensions/System.FloatingPoints.cs:16]: indent

File: src/Qowaiv.Validation.Fluent/FluentValidation/FloatingPointValidation.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
+﻿namespace FluentValidation;
+
+/// <summary>FLuent validation for <see cref="double"/> and <see cref="float"/>.</summary>
+public static class FloatingPointValidation
+{
+    /// <summary>The floating point should be a finite number..</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.Fluent/FluentValidation/FloatingPointValidation.cs:6]: two points

File: src/Qowaiv.Validation.Fluent/Qowaiv.Validation.Fluent.csproj [hunk 9]
```diff
@@ -6,6 +6,8 @@
     <TargetFrameworks>netstandard2.0;net5.0;net6.0;net7.0</TargetFrameworks>
     <Version>0.3.0</Version>
     <PackageReleaseNotes>
+ToBeReleaased
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.Fluent/Qowaiv.Validation.Fluent.csproj:9]: ToBeReleased

File: src/Qowaiv.Validation.Fluent/README.md [hunk 10]
```diff
@@ -1 +1,94 @@
...
+``` C#
+public class CustomValidator : AbstractValidator<Model>
+{
+    public CustomValidator()
+    {
+        RuleFor(m => m.Number).IsFinite();
+    }
+}
+```
+### Email address should be IP-based
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.Fluent/README.md:82]: should **not** be

File: src/Qowaiv.Validation.DataAnnotations/Attributes/MultipleOfAttribute.cs (hunk 5 above)
Comment (Corniel) [reply src/Qowaiv.Validation.DataAnnotations/Attributes/MultipleOfAttribute.cs:33]: After some consideration: a string is an invalid type, it should not be successful.

File: src/Qowaiv.Validation.Fluent/README.md [hunk 11]
```diff
@@ -1 +1,94 @@
...
+``` C#
+public class CustomValidator : AbstractValidator<Model>
+{
+    public CustomValidator()
+    {
+        RuleFor(m => m.Number).IsFinite();
+    }
+}
+```
+### Email address should no be IP-based
```
Comment (LauraXiulan) [inline src/Qowaiv.Validation.Fluent/README.md:82]: no**t**

**Review by LauraXiulan — APPROVED** [review]
